jsonschema-core.xml

```diff
@@ -1476,87 +1508,223 @@
+                                points to the schema of the "additionalProperties" keyword in
+                                the embedded resource.  The canonical URI of that schema, however,
+                                is "https://example.com/bar#/additionalProperties".
+                            </postamble>
+                        </figure>
+                        <figure>
+                            <preamble>
+                                Now consider the following two schema resources linked by reference
+                                using a URI value for "$ref":
+                            </preamble>
+                            <artwork>
+<![CDATA[
+{
+  "$id": "https://example.com/foo",
+  "items": {
+    "$ref": "bar"
+  }
+}
+
+{
+  "$id": "https://example.com/bar",
+  "additionalProperties": { }
+}
+]]>
+                            </artwork>
+                            <postamble>
+                                Here we see that the canonical URI for that "additionalProperties"
+                                subschema is still valid, while the non-canonical URI with the fragment
+                                beginning with "#/items/$ref" now resolves to nothing.
+                            </postamble>
+                        </figure>
                         <t>
-                            To specify such a subschema identifier,
-                            the "$id" keyword is set to a URI reference with a plain name fragment (not a JSON Pointer fragment).
-                            This value MUST begin with the number sign that specifies a fragment ("#"),
-                            then a letter ([A-Za-z]),
-                            followed by any number of letters, digits ([0-9]), hyphens ("-"), underscores ("_"),
-                            colons (":"), or periods (".").
+                            Note also that "https://example.com/foo#/items" is valid in both
+                            arrangments, but resolves to a different value.  This URI ends up
+                            functioning similarly to a retrieval URI for a resource.  While valid,
+                            examining the resolved value and either using the "$id" (if the value
+                            is a subschema), or resolving the reference and using the "$id" of the
+                            reference target, is preferable.
                         </t>
                         <t>
-                            The effect of using a fragment in "$id" that isn't blank or doesn't follow the
-                            plain name syntax is undefined.
+                            An implementation MAY choose not to support addressing schemas
+                            by non-canonical URIs. As such, it is RECOMENDED that schema authors only 
+                            use canonical URIs, as using non-canonical URIs may reduce the 
```

Comment:
```suggestion
                            use canonical URIs, as using non-canonical URIs may reduce
```

Just a gramatical fix on this one. Otherwise, golden.
You can just accept this suggested change =]